uk.co.compucorp.civicrm.hrcomments/tests/phpunit/CRM/HRComments/BAO/CommentTest.php

```diff
@@ -127,4 +150,69 @@ public function testCreatedAtCannotBeChangedWhenUpdatingExistingComment() {
 
     $this->assertEquals($timestampNow, $createdDateTimestamp, '', 10);
   }
+
+  public function testSoftDeleteDoesNotDeleteCommentsFromCommentsTableButSetsIsDeletedFlagToOne() {
+    $comment1 = CommentFabricator::fabricate([
+      'entity_id' => 1,
+      'contact_id' => 1,
+    ]);
+
+    $comment2 = CommentFabricator::fabricate([
+      'entity_id' => 2,
+      'contact_id' => 2,
+    ]);
+
+    Comment::softDelete($comment1->id);
+
+    $comment = new Comment();
+    $comment->find();
+    $this->assertEquals($comment->N, 2);
+
+    $comment->fetch();
+    $this->assertEquals($comment->id, $comment1->id);
+    $this->assertEquals(1, $comment->is_deleted);
+
+    $comment->fetch();
+    $this->assertEquals($comment->id, $comment2->id);
+    $this->assertEquals(0, $comment->is_deleted);
+  }
+
+  public function testIsDeletedAtWillBeZeroIrrespectiveOfTheValuePassedViaIsDeletedParameterOnCreate() {
+    Comment::create([
+      'entity_id' => 1,
+      'entity_name' => 'LeaveRequest',
+      'text' => 'This is a sample comment',
+      'contact_id' => 1,
+      'is_deleted' => 1
+    ]);
+
+    $comment = new Comment();
+    $comment->find();
+    $this->assertEquals($comment->N, 1);
+    $comment->fetch();
+
+    $this->assertEquals(0, $comment->is_deleted);
+  }
+
+  public function testIsDeletedCannotBeChangedWhenUpdatingExistingComment() {
+    $comment1 = Comment::create([
+      'entity_id' => 1,
+      'entity_name' => 'LeaveRequest',
+      'text' => 'This is a sample comment',
+      'contact_id' => 1,
+    ]);
+
+    Comment::create([
+      'id' => $comment1->id,
+      'created_at' => CRM_Utils_Date::processDate('2016-01-01'),
+      'is_deleted' => 1
+    ], false);
+
+    $comment = new Comment();
+    $comment->find();
+    $this->assertEquals($comment->N, 1);
+    $comment->fetch();
+
+    $this->assertEquals(0, $comment->is_deleted);
```

Comment:
On this test, can you try to change some other field and assert that it was not changed? Just to make sure that the updated was fine and only the `is_deleted` value was the one not changed